qa/vagrant/src/main/java/org/elasticsearch/packaging/util/FileUtils.java

```diff
@@ -124,6 +130,41 @@ public static String slurp(Path file) {
         }
     }
 
+    public static String slurpTxtorGz(Path file) {
+        ByteArrayOutputStream fileBuffer = new ByteArrayOutputStream();
+        try (GZIPInputStream in = new GZIPInputStream(Channels.newInputStream(FileChannel.open(file)))) {
+            byte[] buffer = new byte[1024];
+            int len;
+
+            while ((len = in.read(buffer)) != -1) {
+                fileBuffer.write(buffer, 0, len);
+            }
+
+            return (new String(fileBuffer.toByteArray(), StandardCharsets.UTF_8));
+        } catch (ZipException e) {
+            if (e.toString().contains("Not in GZIP format")) {
+                return slurp(file);
+            }
+            throw new RuntimeException(e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static String slurpAllLogs(Path logPath, String activeLogFile, String rotatedLogFilesGlob) {
```

Comment:
We should have at least a minimal javadoc (just a method description) for any public methods added.

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/44423/commits/69b1678433cf040575f554a23f02f4e87b2f68b3